platform-sdk/swirlds-common/src/main/java/com/swirlds/common/wiring/schedulers/internal/ConcurrentTaskScheduler.java

```diff
@@ -74,18 +76,27 @@ public ConcurrentTaskScheduler(
      */
     @Override
     protected void put(@NonNull final Consumer<Object> handler, @NonNull final Object data) {
+        if (getSquelcher().shouldSquelch()) {
+            return;
+        }
+
         onRamp.onRamp();
-        new ConcurrentTask(pool, offRamp, uncaughtExceptionHandler, handler, data).send();
+        new ConcurrentTask(pool, offRamp, uncaughtExceptionHandler, handler, data, getSquelcher()).send();
     }
 
     /**
      * {@inheritDoc}
      */
     @Override
     protected boolean offer(@NonNull final Consumer<Object> handler, @NonNull final Object data) {
-        boolean accepted = onRamp.attemptOnRamp();
+        if (getSquelcher().shouldSquelch()) {
+            // say that the data was accepted, even though it will be thrown into the void
+            return true;
+        }
+
+        final boolean accepted = onRamp.attemptOnRamp();
         if (accepted) {
-            new ConcurrentTask(pool, offRamp, uncaughtExceptionHandler, handler, data).send();
+            new ConcurrentTask(pool, offRamp, uncaughtExceptionHandler, handler, data, getSquelcher()).send();
```

Comment:
If you move squelching to happen in a lambda in bindable input wire, then the tasks will not need to be aware of the squelcher.

Reply:
Done